src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -19,8 +19,39 @@ public static class ConcurrencyUtilities
         // To maintain SHA-1 backwards compatibility with respect to the length of the hex-encoded hash, the hash will be truncated to a length of 20 bytes.
         private const int HashLength = 20;
         private static readonly TimeSpan SleepDuration = TimeSpan.FromMilliseconds(10);
+        private static Mutex GlobalMutex;
+        private static bool UseDeleteOnClose;
         private static readonly KeyedLock PerFileLock = new KeyedLock();
 
+        static ConcurrencyUtilities()
+        {
+            // On Non-Windows platforms we use a global mutex for synchronization.
+            //
+            // On Windows opening and locking the file are performed atomically,
+            // and DeleteOnClose deletes the file when there are no more open handles.
+            //
+            // On Unix this locking is not implemented atomically, and DeleteOnClose
+            // deletes ('unlink') the file as soon as the Handle is closed.
+            //
+            // The following example shows how deleting the file in Process 1
+            // allows both Process 2 and Process 3 to obtain a lock because they open
+            // different file entries.
+            //
+            // Process 1    Process 2   Process 3
+            //   open
+            //   lock         open
+            //   unlink
+            //   unlock                   open
+            //   close
+            //                lock        lock
+
+            UseDeleteOnClose = RuntimeEnvironmentHelper.IsWindows;
+            if (!UseDeleteOnClose)
+            {
+                GlobalMutex = new Mutex(false, "Global\\NuGet.Common.ConcurrencyUtilities");
```

Comment:
@zivkan are you ok with the name of this `Mutex`?

Reply:
yeah, I don't have any better suggestion.